src/ripple/app/ledger/Ledger.cpp

```diff
@@ -372,13 +372,15 @@ getTransaction (Ledger const& ledger,
         txn = Transaction::sharedTransaction (item->peekData (), Validate::YES);
     else if (type == SHAMapTreeNode::tnTRANSACTION_MD)
     {
-        Blob txnData;
-        int txnLength;
-
-        if (!item->peekSerializer().getVL (txnData, 0, txnLength))
-            return Transaction::pointer ();
-
-        txn = Transaction::sharedTransaction (txnData, Validate::NO);
+        try
+        {
+            auto sit = SerialIter{item->data(), item->size()};
+            txn = Transaction::sharedTransaction(sit.getVL(), Validate::NO);
+        }
+        catch (...)
+        {
+            return {};
```

Comment:
More info:  All errors here are fatal.  There are try/catch clauses above here, but none of them catch the types thrown below here.  A throw from results in program termination via uncaught exception.  Furthermore, the one place that calls this code assumes that the pointer returned is always non-null.  If we return a nullptr here, we will get a null-dereference crash.